context/app/static/js/components/Detail/FileBrowser/FileBrowser.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect } from 'react';
+import FileBrowserNode from '../FileBrowserNode';
+import { ScrollPaper } from './style';
+
+function FileBrowser(props) {
+  const { files } = props;
+  const [fileTree, setFileTree] = useState({});
+  useEffect(() => {
```

Comment:
Minor thing but if possible, it would be nice to extract this logic in the useEffect to a utility function stored in a separate js file.